crates/worker/src/partition/services/non_deterministic/remote_context.rs

```diff
@@ -479,16 +459,33 @@ impl<'a, State: StateReader> InvocationContext<'a, State> {
         &mut self,
         operation_id: &str,
         retention_period_sec: u32,
-        entry_index: EntryIndex,
-        result: ResponseResult,
+        virtual_journal_length: EntryIndex,
     ) -> Result<(), InvocationError> {
         let expiry_time = SystemTime::now() + Duration::from_secs(retention_period_sec as u64);
+        let journal_service_id = self.journal_service_id();
+
+        // Load the result from the journal last entry
+        let output_stream_entry_index = virtual_journal_length - 1;
+        let entry = self
+            .read_journal_entry(&journal_service_id, output_stream_entry_index)
+            .await?
+            .ok_or_else(|| {
+                InvocationError::internal(format!(
+                    "expected entry at index {}",
+                    output_stream_entry_index
+                ))
+            })?;
+        let_assert!(
+            Entry::OutputStream(OutputStreamEntry { result }) = entry
+                .deserialize_entry::<ProtobufRawEntryCodec>()
+                .map_err(InvocationError::internal)?
+        );
```

Comment:
Could remembering the result of an invocation (not setting its state to `Done`) happen when seeing the first `OutputStreamEntry` in the `handle_entry` method? That way, it is in line with how the PP works and the code would already support embedded handlers that defer their response (this probably only makes sense once we have keyed embedded handlers tbh).

Reply:
I think deferred responses for the embedded handler API does not make sense to have. So this is probably fine to keep as it is.

Reply:
> So this is probably fine to keep as it is.

True, but how the SDK models this is also unclear (e.g. this comment https://github.com/restatedev/sdk-typescript/issues/183#issuecomment-1851435355). Hence I would like the behavior to be as close as possible to the invoker.

Reply:
Also If we start using this EndMessage for other stuff (e.g. metadata about the invocation), then even more we should have an aligned behavior.